File: src/class/Vino.js
export default class Vino {
  constructor(
    añada,
    imagenEtiqueta,
    nombre,
    notaDeCataBodega,
    precioARS,
    varietal,
    maridaje
  ) {
    this.añada = añada;
    this.imagenEtiqueta = imagenEtiqueta;
    this.nombre = nombre;
    this.notaDeCataBodega = notaDeCataBodega;
    this.precioARS = precioARS;
    this.varietal = varietal; // Relación con la clase Varietal
    this.maridaje = maridaje; // Relación con la clase maridaje
  }

  // Métodos getter y setter para los atributos de la clase Vino

  get añada() {
    return this.añada;
  }

  set añada(añada) {
    this.añada = añada;
  }

  get imagenEtiqueta() {
    return this.imagenEtiqueta;
  }

  set imagenEtiqueta(imagenEtiqueta) {
    this.imagenEtiqueta = imagenEtiqueta;
  }

  get nombre() {
    return this.nombre;
  }

  set nombre(nombre) {
    this.nombre = nombre;
  }

  get notaDeCataBodega() {
    return this.notaDeCataBodega;
  }

  set notaDeCataBodega(notaDeCataBodega) {
    this.notaDeCataBodega = notaDeCataBodega;
  }

  get precioARS() {
    return this.precioARS;
  }

  set precioARS(precioARS) {
    this.precioARS = precioARS;
  }

  get varietal() {
    return this.varietal;
  }

  set varietal(varietal) {
    this.varietal = varietal;
  }

  get maridaje() {
    return this.maridaje;
  }

  set maridaje(maridaje) {
    this.maridaje = maridaje;
  }
}

File: src/com/BonVino/proyect/Vino.java
package com.BonVino.proyect;

public class Vino {
    private int añada;
    private String imagenEtiqueta;
    private String nombre;
    private String notaDeCataBodega;
    private double precioARS;
    private Varietal varietal; // Relación con la clase Varietal
    private Maridaje maridaje; // Relación con la clase Maridaje

    // Constructor
    public Vino(int añada, String imagenEtiqueta, String nombre, String notaDeCataBodega,
                double precioARS, Varietal varietal, Maridaje maridaje) {
        this.añada = añada;
        this.imagenEtiqueta = imagenEtiqueta;
        this.nombre = nombre;
        this.notaDeCataBodega = notaDeCataBodega;
        this.precioARS = precioARS;
        this.varietal = varietal;
        this.maridaje = maridaje;
    }

    // Métodos getter y setter para añada
    public int getAñada() {
        return añada;
    }

    public void setAñada(int añada) {
        this.añada = añada;
    }

    // Métodos getter y setter para imagenEtiqueta
    public String getImagenEtiqueta() {
        return imagenEtiqueta;
    }

    public void setImagenEtiqueta(String imagenEtiqueta) {
        this.imagenEtiqueta = imagenEtiqueta;
    }

    // Métodos getter y setter para nombre
    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    // Métodos getter y setter para notaDeCataBodega
    public String getNotaDeCataBodega() {
        return notaDeCataBodega;
    }

    public void setNotaDeCataBodega(String notaDeCataBodega) {
        this.notaDeCataBodega = notaDeCataBodega;
    }

    // Métodos getter y setter para precioARS
    public double getPrecioARS() {
        return precioARS;
    }

    public void setPrecioARS(double precioARS) {
        this.precioARS = precioARS;
    }

    // Métodos getter y setter para varietal
    public Varietal getVarietal() {
        return varietal;
    }

    public void setVarietal(Varietal varietal) {
        this.varietal = varietal;
    }

    // Métodos getter y setter para maridaje
    public Maridaje getMaridaje() {
        return maridaje;
    }

    public void setMaridaje(Maridaje maridaje) {
        this.maridaje = maridaje;
    }
}
